src/transaction/transaction.spec.ts

```diff
@@ -150,6 +153,516 @@ describe('transaction', () => {
+
+      appFactory.send.create
+
+      await appClient1.fundAppAccount({ amount: algo(2) })
+      await appClient2.fundAppAccount({ amount: algo(2) })
+      await appClient3.fundAppAccount({ amount: algo(2) })
+    })
+
+    test('throws when no max fee is supplied', async () => {
+      const params = {
+        method: 'no_op',
+        coverAppCallInnerTransactionFees: true,
+      } satisfies Parameters<(typeof appClient1)['send']['call']>[0]
+
+      await expect(async () => await appClient1.send.call(params)).rejects.toThrow(
+        'Please provide a maxFee for each app call transaction when coverAppCallInnerTransactionFees is enabled. Required for transaction 0',
+      )
+    })
+
+    test('throws when inner transaction fees are not covered and coverAppCallInnerTransactionFees is disabled', async () => {
+      const expectedFee = 7000n
+      const params = {
+        method: 'send_inners_with_fees',
+        args: [appClient2.appId, appClient3.appId, [0n, 0n, 0n, 0n, [0n, 0n]]],
+        maxFee: microAlgo(expectedFee),
+        coverAppCallInnerTransactionFees: false,
+      } satisfies Parameters<(typeof appClient1)['send']['call']>[0]
+
+      await expect(async () => await appClient1.send.call(params)).rejects.toThrow(/fee too small/)
+    })
+
+    test('does not alter fee when app call has no inners', async () => {
+      const expectedFee = 1000n
+      const params = {
+        method: 'no_op',
+        coverAppCallInnerTransactionFees: true,
+        maxFee: microAlgo(2000),
+      } satisfies Parameters<(typeof appClient1)['send']['call']>[0]
+      const result = await appClient1.send.call(params)
+
+      expect(result.transaction.fee).toBe(expectedFee)
+      await assertMinFee(appClient1, params, expectedFee)
+    })
+
+    test('throws when max fee is too small to cover inner transaction fees', async () => {
+      const expectedFee = 7000n
+      const params = {
+        method: 'send_inners_with_fees',
+        args: [appClient2.appId, appClient3.appId, [0n, 0n, 0n, 0n, [0n, 0n]]],
+        maxFee: microAlgo(expectedFee - 1n),
+        coverAppCallInnerTransactionFees: true,
+      } satisfies Parameters<(typeof appClient1)['send']['call']>[0]
+
+      await expect(async () => await appClient1.send.call(params)).rejects.toThrow(
+        /Fees were too small to resolve execution info via simulate/,
+      )
+    })
+
+    test('throws when static fee is too small to cover inner transaction fees', async () => {
+      const expectedFee = 7000n
+      const params = {
+        method: 'send_inners_with_fees',
+        args: [appClient2.appId, appClient3.appId, [0n, 0n, 0n, 0n, [0n, 0n]]],
+        staticFee: microAlgo(expectedFee - 1n),
+        coverAppCallInnerTransactionFees: true,
+      } satisfies Parameters<(typeof appClient1)['send']['call']>[0]
+
+      await expect(async () => await appClient1.send.call(params)).rejects.toThrow(
+        /Fees were too small to resolve execution info via simulate/,
```

Comment:
I think this error should be a bit clearer.

Reply:
The full message is "Fees were too small to resolve execution info via simulate. You may need to increase an app call transaction maxFee.". Do you think this is clear enough?